std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,77 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Finds the mean (colloquially known as the average) of a range.
+Is $(D O(r.length)).
+
+Params:
+    r = a input range
+
+Returns:
+    the mean of r
+*/
+auto mean(R)(R r) if (isInputRange!R && !isInfinite!R) {
+    import std.typecons : tuple;
+
+    immutable auto pair = reduce!((a, b) => tuple(a[0] + 1, a[1] + b))(tuple(0u, 0.0), r);
+    return pair[1] / pair[0];
+}
+
+///
+@safe unittest {
+    assert(mean([1, 2, 3]) == 2);
```

Comment:
Do you mean there would be a problem with floating point imprecision? How else do you check for equality?